src/main/java/org/kaazing/netx/ws/internal/DefaultWebSocketFactory.java

```diff
@@ -81,126 +92,61 @@ public WebSocket createWebSocket(URI location)
-        }
-
-        // Clone the map of default parameters maintained at the
-        // WebSocketFactory level to pass into the WebSocket instance.
-        Map<String, WebSocketExtensionParameterValues> enabledParams =
-                      new HashMap<String, WebSocketExtensionParameterValues>();
-        enabledParams.putAll(_parameters);
-
         // Create a WebSocket instance that inherits the enabled protocols,
         // enabled extensions, enabled parameters, the HttpRedirectOption,
         // the extension factories(ie. the supported extensions).
-        WebSocketImpl   ws = new WebSocketImpl(location, enabledParams);
-        ws.setRedirectPolicy(_redirectPolicy);
-//        ws.setEnabledExtensions(enabledExtensions);
-        ws.setEnabledProtocols(enabledProtocols);
-        ws.setChallengeHandler(_challengeHandler);
-        ws.setConnectTimeout(_connectTimeout);
+        WebSocketImpl   ws = new WebSocketImpl(location);
+        ws.setRedirectPolicy(redirectPolicy);
+        ws.setEnabledProtocols(protocols);
+        ws.setChallengeHandler(challengeHandler);
+        ws.setConnectTimeout(connectTimeout);
+
+        for (WebSocketExtension extension : enabledExtensions) {
+            ws.addEnabledExtension(extension);
+        }
 
         return ws;
     }
 
     @Override
     public int getDefaultConnectTimeout() {
-       return _connectTimeout;
+       return this.connectTimeout;
     }
 
     @Override
     public ChallengeHandler getDefaultChallengeHandler() {
-        return _challengeHandler;
+        return this.challengeHandler;
     }
 
     @Override
-    public Collection<String> getDefaultEnabledExtensions() {
-        return (_enabledExtensions == null) ? Collections.<String>emptySet() :
-                                              unmodifiableCollection(_enabledExtensions);
+    public Collection<WebSocketExtension> getDefaultEnabledExtensions() {
+        return this.enabledExtensionsRO;
     }
 
     @Override
     public HttpRedirectPolicy getDefaultRedirectPolicy() {
-        return _redirectPolicy;
-    }
-
-    @Override
-    public <T> T getDefaultParameter(Parameter<T> parameter) {
-        String                            extName = parameter.extension().name();
-        WsExtensionParameterValuesImpl paramValues = _parameters.get(extName);
-
-        if (paramValues == null) {
-            return null;
-        }
-
-        return paramValues.getParameterValue(parameter);
+        return redirectPolicy;
     }
 
     @Override
     public Collection<String> getSupportedExtensions() {
-        return (_supportedExtensions == null) ? Collections.<String>emptySet() :
-                                                unmodifiableCollection(_supportedExtensions);
+        return (this.supportedExtensions == null) ? Collections.<String>emptySet() :
+                                                unmodifiableCollection(this.supportedExtensions);
```

Comment:
Recommend hoisting empty set to constant to avoid allocation per call.
Also suggest pre-creating `supportedExtensionsRO` to avoid allocation per call.